### PR TITLE
[admin] index update destroy user

### PR DESCRIPTION
**Pull request by ngtrduc**

### CODE REVIEW

File: app/controllers/admin/profiles_controller.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      flash[:success] = t :profile_updated
+      respond_to do |format|
+        format.html {redirect_to [:admin, @profile.user]}
+        format.js
+      end
+    else
+      redirect_to [:admin, @profile.user]
+    end
+	end
+
```
Comment (qtsoul95) [inline app/controllers/admin/profiles_controller.rb]: indent

File: app/views/admin/users/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+          <td><%= user.email %></td>
+          <td><%= user.created_at.strftime t(:time_type) %></td>
+          <td><%= user.updated_at.strftime t(:time_type) %></td>
+          <td class="dn-action-padding">
+            <%= link_to t(:show_profile), admin_user_path(user),
+              class: "btn btn-primary" %>
+            <% unless current_user?(user) %>
+              <%= link_to t(:delete), [:admin, user], method: :delete, 
+                data: {confirm: t(:sure)}, class: "btn btn-danger btn-primary" %>
+            <% end %>
```
Comment (qtsoul95) [inline app/views/admin/users/index.html.erb]: <%= link_to(t(:delete), [:admin, user], method: :delete, data: {confirm: t(:sure)}, class: "btn btn-danger btn-primary") unless curent_user? user %>

File: app/controllers/admin/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+class Admin::UsersController < ApplicationController
+  load_and_authorize_resource
+
+  def index
+    @users = @users.paginate(page: params[:page]).per_page Settings.page_size
+  end
+  
+  def show
+    @profile = @user.profile
+    @positions = Position.all
```
Comment (hellovietnam93) [inline app/controllers/admin/users_controller.rb]: sử dụng metaprogramming nhé, instance_variable_set

File: app/models/ability.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
```
Comment (hellovietnam93) [inline app/models/ability.rb]: bỏ ()

File: app/controllers/admin/users_controller.rb [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+  def destroy
+    @user.destroy
+    flash[:success] = t :destroy_success
+    redirect_to admin_users_path
+  end
+
+  private
+  def get_data
+    self.instance_variable_set(:@positions, Position.all)
```
Comment (hellovietnam93) [inline app/controllers/admin/users_controller.rb]: [Position, Division, Skill].each do |model
  instance_variable_set "@#{model.table_name}", model.all
end

File: app/views/admin/profiles/update.js.erb [hunk 6]
```diff
@@ -0,0 +1,6 @@
+$('.edit-form').addClass('hidden')
```
Comment (hellovietnam93) [inline app/views/admin/profiles/update.js.erb]: double quote

File: app/views/admin/users/show.html.erb [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+        <%= f.collection_select :division_id, @divisions, :id, :name,
+          {prompt: "Select a Division"}, {class: "edit-form hidden"} %>
+        <span id="p-division">
+          <%= @profile.division.name unless @profile.division.nil? %>
+        </span>
+      </p> <br/>
+
+      <p><%= t :skill %>: 
+        <%= f.collection_select :skill_id, @skills, :id, :name,
+          {prompt: "Select a Skill"}, {class: "edit-form hidden"} %>
```
Comment (hellovietnam93) [inline app/views/admin/users/show.html.erb]: I18n

**Pull request merged**